zkevm-circuits/src/evm_circuit/execution/callop.rs

```diff
@@ -54,32 +57,27 @@ pub(crate) struct CallOpGadget<F> {
 }
 
 impl<F: Field> ExecutionGadget<F> for CallOpGadget<F> {
-    const NAME: &'static str = "CALL_STATICCALL";
+    const NAME: &'static str = "CALL_DELEGATECALL_STATICCALL";
 
-    const EXECUTION_STATE: ExecutionState = ExecutionState::CALL_STATICCALL;
+    const EXECUTION_STATE: ExecutionState = ExecutionState::CALL_DELEGATECALL_STATICCALL;
 
     fn configure(cb: &mut ConstraintBuilder<F>) -> Self {
         let opcode = cb.query_cell();
         cb.opcode_lookup(opcode.expr(), 1.expr());
-        let is_call = (OpcodeId::STATICCALL.expr() - opcode.expr())
-            * F::from(OpcodeId::STATICCALL.as_u64() - OpcodeId::CALL.as_u64())
-                .invert()
-                .unwrap();
+        let is_call = is_opcode(&opcode, OpcodeId::CALL);
```

Comment:
better to use `IsZero` gadget here. otherwise the degree of the lagrange-based polynomial will be too high, especially after you add `CALLCODE` in this gadget.

Reply:
Fixed to replace `is_call`, `is_delegatecall` and `is_staticcall` as:
```
let is_call = IsZeroGadget::construct(cb, opcode.expr() - OpcodeId::CALL.expr());
let is_delegatecall = IsZeroGadget::construct(cb, opcode.expr() - OpcodeId::DELEGATECALL.expr());
let is_staticcall = 1.expr() - is_call.expr() - is_delegatecall.expr();
```
And deleted the below constraint:
```
cb.require_equal(
    "Opcode should be CALL, DELEGATECALL or STATICCALL",
    is_call.expr() + is_delegatecall.expr() + is_staticcall.expr(),
    1.expr(),
);
```